src/javascript-wrapper.js

```diff
@@ -44,7 +44,12 @@ var fRunUpdater = function(updater, updaterFormals, vni) {
              if (newStateLm === oldOutputStateLm ||
                  _.isUndefined(oldOutputStateLm) && ! _.isUndefined(newStateLm)) {
                  // updater did not modify output state lm - update it now
-                 vni.outputState({data: results, groupLm: groupLm, lm: createLm()});
+                 if (vni.nodeInstance.isTranslator && ! _.isUndefined(results.id)) {
+                     var id = 'urn:local:'+results.id+":(translator:"+vni.nodeInstance.componentName+")"; 
```

Comment:
Small request: I think spaces around the '+' operators would make the above line a bit easier for my eyes to parse.